examples/src/main/java/tachyon/examples/BasicNonByteBufferOperations.java

```diff
@@ -83,32 +108,33 @@ private void write(TachyonFS client, TachyonURI filePath, WriteType writeType,
     }
   }
 
-  private TachyonFile getOrCreate(TachyonFS client, TachyonURI filePath, boolean deleteIfExists)
-      throws IOException {
+  private FileOutStream getOrCreate(TachyonFileSystem tachyonFileSystem, TachyonURI filePath,
+      boolean deleteIfExists, OutStreamOptions clientOptions) throws IOException, TachyonException {
     TachyonFile file;
+
     try {
-      file = client.getFile(filePath);
-    } catch (IOException ioe) {
+      file = tachyonFileSystem.open(filePath);
+    } catch (Exception e) {
       file = null;
     }
     if (file == null) {
       // file doesn't exist yet, so create it
-      long fileId = client.createFile(filePath);
-      file = client.getFile(fileId);
+      return tachyonFileSystem.getOutStream(filePath, clientOptions);
     } else if (deleteIfExists) {
       // file exists, so delete it and recreate
-      client.delete(new TachyonURI(file.getPath()), false);
-
-      long fileId = client.createFile(filePath);
-      file = client.getFile(fileId);
+      tachyonFileSystem.delete(file);
+      return tachyonFileSystem.getOutStream(filePath, clientOptions);
     }
-    return file;
+    // file exists and deleteIfExists is false
+    throw new FileAlreadyExistsException(" file exists but deleteIfExists is false");
```

Comment:
@apc999 because in TachyonFileSystem we're not supposed to get a output stream for an already exist file, I guess we'd better throw an exception here.